client/src/app/containers/Alert/useAlert.ts

```diff
@@ -0,0 +1,32 @@
+import { nanoid } from '@reduxjs/toolkit';
+import { useDispatch } from 'react-redux';
+
+import { actions } from './slice';
+import { AlertType } from './types';
+
+export interface AlertSetter {
+    (message: string, type: AlertType): string;
```

Comment:
:mag: :book: is it possible/helpful to add the name of the returned string (i.e. `alertID`)?

and just to ensure I'm on the same page: if I were to express what's happening here in my own words, it's an interface which is *just a single function* that (1) accepts a message string and an AlertType (which is a string too, but it's been typed to essentially be an enum) and (2) returns an alertID. What did I get right/wrong?

Reply:
Again, don't think we need this anymore. However...

Your description is mostly right. `AlertSetter` essentially defines the signature for a function that takes the arguments as you described (`AlertType` is a union type which isn't _exactly_ an enum). The ID is just a string; there aren't really named returns like there are in go (and I think most of the time they aren't helpful?).